src/main/java/com/ibm/cloud/cloudant/kafka/mappers/SinkRecordToDocument.java

```diff
@@ -78,38 +81,37 @@ private Map<String, Object> convertStruct(Struct struct, Map<String, Object> out
         // iterate fields and add to map
         for (Field f : schema.fields()) {
             Object value = struct.get(f);
-            outMap.put(f.name(), getField(f.schema().type(), value));
+            outMap.put(f.name(), convertItemFromStruct(f.schema().type(), value));
         }
         return outMap;
     }
 
     // convert kafka map to map by adding key/values to passed in map, and returning it
     private Map<String, Object> convertMap(Map<?,?> inMap, Map<String, Object> outMap) {
 
+        // iterate over keys and add to map
         for (Object k : inMap.keySet()) {
             if (k instanceof String) {
                 Object v = inMap.get(k);
```

Comment:
My understanding is that it is more efficient to use the entry set if `get` needs to be called every time for the value.
```suggestion
        for (Map.Entry<?, ?> entry : inMap.entrySet()) {
            if (entry.getKey() instanceof String) {
```